### PR TITLE
Upgrade jax and jaxlib, install with conda

### LINKED ISSUE #583: Install jax with conda
See #582 for context.

When (pip-installed) `jax` is loaded first, the system's `libstdc++.so` is cached, and `rdkit` ends up using that version, which is potentially older than it expects. `rdkit` should load the `libstdc++.so` from the conda env regardless of import order.

Several potential solutions were discussed:
1. Include the conda library dir in `LD_LIBRARY_PATH` so jax loads conda's `libstdc++.so`
2. Install `jax` and `jaxlib` with conda so they link to conda's library dir
3. Upgrade the CI environment to Ubuntu 20.04
4. Upgrade `glibc` in the existing CI env (suggested by @bp-kelley)

### PR DESCRIPTION
**Pull request by mcwitt**
Supersedes #539, closes #583.

- Upgrades jax and jaxlib versions to latest
- Pulls in changes from #539 needed for compatibility with newer jax versions (b5e3c63)
- Updates config and README instructions to install jax(lib) with conda (see #583 for motivation)
- Adds `environment.yml`

#### Todo
- [x] Update `timemachine-ansible` [[PR]](https://git.relaytxinternal.com/fyork/timemachine-ansible/-/merge_requests/9)

### CODE REVIEW

File: fe/loss.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def mybar_impl(w):
     return A
 
 
-def mybar_vjp(g, w):
+def mybar_jvp(g, w):
```
Comment (mcwitt) [inline fe/loss.py:24]: This was previously discussed starting [here](https://github.com/proteneer/timemachine/pull/539#discussion_r754718532).

File: README.md [hunk 2]
```diff
@@ -38,9 +38,9 @@ We currently support the following functional forms. Parameters that can be opti
 If using conda the following can be used to configure your environment
 
 ```
-conda create -n timemachine python=3.7
+conda env create -f environment.yml -n timemachine
+conda install openmm=7.5.1 -c conda-forge # only if using openmm from conda
```
Comment (mcwitt) [inline README.md:42]: Sometimes we want to use `openmm` built from source, so I opted to leave it out of `environment.yml` for flexibility.

**Review by proteneer — APPROVED** [review]
lgtm  - though not a big fan of the defvjp vs defjvp luckily coinciding but its technically correct.

File: README.md (hunk 2 above)
Comment (maxentile) [reply README.md:42]: Ahh, I've been using `openmm=7.7` -- reminds me we might later want to consider pinning to an OpenMM version more recent than the simtk unit namespace migration.

**Review by maxentile — APPROVED** [review]
Looks good!

**Pull request merged**